### ISSUE TITLE
Backslash escapes are duplicated during serialization in some cases

### ISSUE THREAD

**Issue by marijnh** [ProseMirror/prosemirror-markdown#3]
Via https://github.com/ProseMirror/prosemirror/issues/562:

It sounds like a feature, but it's a bug :1st_place_medal: 

# Repro

http://prosemirror.net/demo/markdown.html

Type this into the Markdown editor 

```
`~`
```

Switch between WYSIWYG and Markdown mode

# Result

![image](https://cloud.githubusercontent.com/assets/856194/23006947/07caab56-f459-11e6-8984-25fe706ff391.png)

**Comment by torifat** [comment]
It's because of this -> https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.js#L228

There's no `isCode` in `code` mark.

**Comment by marijnh** [comment]
The information of whether to escape inside a given mark should probably be put in the parser spec, rather than the schema, anyway (since it's Markdown-specific).

**Comment by mhuebert** [comment]
Just to see if the rest of the implementation is working, I've tried adding an `isCode` attribute (true) on the code mark, and it doesn't appear to resolve the issue. But I might be missing something w.r.t. how it is supposed to work.

**Comment by Gozala** [comment]
> The information of whether to escape inside a given mark should probably be put in the parser spec, rather than the schema, anyway (since it's Markdown-specific).

@marijnh I would like to fix this, but I'm not sure I understand your suggestion here. It does not seems to be markdown parser issue but rather of markdown serializer which does escape text with in the code blocks. I also think issue is here:

https://github.com/ProseMirror/prosemirror-markdown/blob/39224ecc4a043cf84834b7ec292d87e08031f6fb/src/to_markdown.js#L240

As far as I can tell there is no way to make `isCode` true there.

**Comment by Gozala** [comment]
Here is the solution I would like to propose:

1. Extend `MarkSpec` interface to add `code?: boolean` filed [similar to one in `NodeSpec`](http://prosemirror.net/docs/ref/#model.NodeSpec.code)
2. Add `isCode` boolean field to both [`MarkType`](https://github.com/ProseMirror/prosemirror-model/blob/cdded3da5d0a21f1091f6eb6232d568af4a8e76d/src/schema.js#L236)  and [`NodeType`](https://github.com/ProseMirror/prosemirror-model/blob/cdded3da5d0a21f1091f6eb6232d568af4a8e76d/src/schema.js#L46) that would be set assigned as `this.isCode = spec.code === true` in the constructor.
3. Add `code:true` to the [`code` mark in the schema](https://github.com/ProseMirror/prosemirror-markdown/blob/39224ecc4a043cf84834b7ec292d87e08031f6fb/src/schema.js#L141-L144)

Alternatively I could just add `this.isCode = name == "code"` in `MarkType` constructor which would be similar to how `isText` is set on [`NodeType`](https://github.com/ProseMirror/prosemirror-model/blob/cdded3da5d0a21f1091f6eb6232d568af4a8e76d/src/schema.js#L84), although I would prefer more general approach described above since I'm using some custom markdown extensions that this would not cover.

Yet another and least favorite option would be to substitute following line:
https://github.com/ProseMirror/prosemirror-markdown/blob/39224ecc4a043cf84834b7ec292d87e08031f6fb/src/to_markdown.js#L240

With

```js
 let code = marks.length && marks[marks.length - 1].type.spec.isCode && marks[marks.length - 1] 
```

That would require no changes to `prosemirror-model` but seems dirty since, `MarkSpec` interface defines `isCode` field.

**Comment by marijnh** [comment]
> It does not seems to be markdown parser issue but rather of markdown serializer

Yeah, sorry, I think I meant serializer when I wrote that.

> Extend MarkSpec interface to add code?: boolean filed similar to one in NodeSpec

No, as I mentioned, I'd prefer for this to stay in the serializer, since it's Makdown-specific and doesn't need to concern the schema.

**Comment by susnux** [comment]
It seems this was fixed in version 1.2.0, with https://github.com/ProseMirror/prosemirror-markdown/commit/eff71629d88fb93007aa33f32de4b8d043b19b7d and can be closed (?)
